packages/twenty-server/src/database/commands/upgrade-version-command/1-8/1-8-migrate-attachment-author-to-created-by.command.ts

```diff
@@ -0,0 +1,140 @@
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Command } from 'nest-commander';
+import { Repository } from 'typeorm';
+
+import {
+  ActiveOrSuspendedWorkspacesMigrationCommandRunner,
+  type RunOnWorkspaceArgs,
+} from 'src/database/commands/command-runners/active-or-suspended-workspaces-migration.command-runner';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { FieldActorSource } from 'src/engine/metadata-modules/field-metadata/composite-types/actor.composite-type';
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { STANDARD_OBJECT_IDS } from 'src/engine/workspace-manager/workspace-sync-metadata/constants/standard-object-ids';
+
+@Command({
+  name: 'upgrade:1-8:1-8-migrate-attachment-author-to-created-by',
+  description:
+    'Migrate attachment author relation to createdBy composite field',
+})
+export class MigrateAttachmentAuthorToCreatedByCommand extends ActiveOrSuspendedWorkspacesMigrationCommandRunner {
+  constructor(
+    @InjectRepository(Workspace)
+    protected readonly workspaceRepository: Repository<Workspace>,
+    protected readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    @InjectRepository(ObjectMetadataEntity)
+    private readonly objectMetadataRepository: Repository<ObjectMetadataEntity>,
+  ) {
+    super(workspaceRepository, twentyORMGlobalManager);
+  }
+
+  override async runOnWorkspace({
+    workspaceId,
+    dataSource: _dataSource,
+  }: RunOnWorkspaceArgs): Promise<void> {
+    // Find the attachment object metadata
+    const attachmentObjectMetadata =
+      await this.objectMetadataRepository.findOne({
+        where: {
+          workspaceId,
+          standardId: STANDARD_OBJECT_IDS.attachment,
+        },
+      });
+
+    if (!attachmentObjectMetadata) {
+      this.logger.log(
+        `Attachment object metadata not found for workspace ${workspaceId}, skipping migration`,
+      );
+
+      return;
+    }
+
+    this.logger.log(
+      `Migrating attachment author to createdBy for workspace ${workspaceId}`,
+    );
+
```

Comment:
Could you please also deprecated the logic we had to fill the author? (let's not deprecate the author field yet, we'll deprecate it in an upcoming release)